src/scene_battle_rpg2k3.cpp

```diff
@@ -427,9 +428,33 @@ void Scene_Battle_Rpg2k3::UpdateAnimations() {
 					enemy_cursor->SetX(enemy->GetBattlePosition().x + sprite->GetWidth() / 2 + 2);
 					enemy_cursor->SetY(enemy->GetBattlePosition().y - enemy_cursor->GetHeight() / 2);
 
-					auto* state = enemy->GetSignificantState();
-					if (state) {
-						help_window->SetText(ToString(state->name), state->color);
+					std::vector<lcf::rpg::State*> ordered_states;
+					const auto& states = enemy->GetStates();
+					for (size_t i = 0; i < states.size(); ++i) {
+						if (states[i] > 0) {
+							lcf::rpg::State* state = lcf::ReaderUtil::GetElement(lcf::Data::states, i + 1);
+							ordered_states.push_back(state);
+						}
+					}
```

Comment:
This for loop here (which basicly fetches the objects for state ids) should be also a helper function in the state class.

Something like ``std::vector<State*> GetObjects()``

Reply:
Added.